src/connectors/configure/connectConfigure.ts

```diff
@@ -78,18 +94,23 @@ const connectConfigure: ConfigureConnector = function connectConfigure(
 
     function refine(helper: AlgoliaSearchHelper): Refine {
       return (searchParameters: PlainSearchParameters) => {
+        const newSearchParameters = excludeInvalidUserToken(searchParameters);
```

Comment:
hmm, I think instead this should be in a dev check only, we can't validate all search parameters in configure. There's other invalid combinations, but those get rejected by the API.

Reply:
Many customers especially on ecommerce don't use bundlers, and they don't get dev warnings. And this must be something more than a warning, because it will definitely screw the analysis. However, I don't think it's something the API should "reject", because on their end, they just receive a string `"null"` which they just cannot reject or ignore. It's on our side to send a correct one, IMO.

Reply:
I'm not sure this is the right place for this check though, can't you send userToken with different search parameter methods as well?

Reply:
> send userToken with different search parameter methods

can you explain a little more, especially 'different search parameter methods' part?

Reply:
you could set it with the helper too, or with configure related items, or with voice search. I'm not sure if this is the right place

Reply:
Ah I've got your point. It makes sense. To block the invalid token fundamentally, we should do it in helper. What do you think?

Reply:
I guess so, but aren't there other invalid values than null & ""? Won't this become a long list of values eventually?

Reply:
I just checked and
* insights API: throws an error with invalid token
* search API: doesn't throw

So I guess we should not send any invalid user token to search API because there doesn't seem to be a guard. The rule is simple, though: `[a-zA-Z0-9_-]{1,64}` (https://www.algolia.com/doc/rest-api/insights/#method-param-usertoken)

Reply:
Like @Haroenv I think we should do this check in the helper (if we agree that it is the helper responsibility to validate the parameters) and throw an exception in it in case of an invalid token (with an helpful error message to ease debugging)

Reply:
@yannickcr which one do you think is proper? exception vs warning
I feel like it's a bit too much to throw an exception.
My rationale is 

> The user token is wrong? okay, this request won't be analyzed. But let's still provide the search result.

Reply:
Does sending an invalid token to the search API has any impact ? If no then a warning should suffice imho.

Reply:
the problem is that if we throw an error, it will break their search interface, for something inconsequential to the actual search.

Reply:
Yes, that's why I said a warning should be enough if an invalid userToken has no impact :wink: